server/parsec/asgi/rpc.py

```diff
@@ -54,6 +54,32 @@
 logger = get_logger()
 
 
+def block_repr(block: bytes) -> str:
+    if len(block) <= 64:
+        return repr(block)
+    return f"{block[:64]}... ({len(block)} bytes)"
+
+
+@dataclass
+class LoggedReq:
+    req: object
+
+    def __repr__(self) -> str:
+        if not isinstance(self.req, authenticated_cmds.latest.block_create.Req):
+            return repr(self.req)
+        return f"BlockCreateReq {{ block: {block_repr(self.req.block)} }}"
```

Comment:
Maybe redundant but how about adding the comments you put in the PR description? 
Also I would reverse condition (matter of taste ^^)
```suggestion
        # block_create has a large payload (up to 512K)
        # truncate it to save space in logs 
        if isinstance(self.req, authenticated_cmds.latest.block_create.Req):
            return f"BlockCreateReq {{ block: {block_repr(self.req.block)} }}"
        return repr(self.req)
```

Reply:
4556e098267f7fb6060d38048074439c1100a41f